enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,894 @@
+```
+
+Check if the router-default service was created with proper annotations.
+
+```yaml
+% oc get svc router-default -n openshift-ingress -o yaml         
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-0956fea34de4cb7ab,eipalloc-0e9a3077a70de050a,eipalloc-0b69fc4691f54cdd0,eipalloc-01e6ba6cbba1a391b,eipalloc-0e242df173f906112
+...
+  type: LoadBalancer
+status:
+  loadBalancer:
+    ingress:
+    - hostname: a8491585620154d5992dcdecc06c5779-7938285b62f7bebe.elb.us-east-1.amazonaws.com
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+The process in general works as follows:
+- cluster-admin sets spec.
+- desiredLoadBalancerService sets the annotation based on spec.
+- syncIngressControllerStatus sets status based on the annotation.
+
+If the cluster-admin manually sets the annotation, then in that case syncIngressControllerStatus will reflect it to status.
+More details on the implementation are as follows:
+
+#### 1. Create a new IngressController with EIPs:
+
+   The admin sets the `eipAllocations` in the `IngressController` CR. The cluster ingress operator then creates a `IngressController` by:
+   1. scaling a deployment for the `IngressController` CR which has the value of the `eipAllocations` from the
+      ingress cluster object.
+   2. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+      which uses the value from the field `eipAllocations` of `IngressController` CR.  
+
+#### 2. Update the EIP for an existing IngressController:
+   ##### Effectuating EIP Updates using manual method:
+   When the admin changes the eipAllocations field in the `IngressController` CR, the ingress operator sets the Load Balancer
+   Progressing status as per [section](#cluster-administrator-updates-eipallocations-field-of-the-existing-ingress-controller).
+   The cluster admin has to manually delete service in order to effectuate the change.
+   Then step 1.2 is followed by the ingress operator.
+   We have intentionally designed this feature to require manual deletion of the service for the following reasons:
+
+   Reason #1: To mitigate risks associated with cluster admins providing an invalid annotation value.
+   While the AWS Cloud Controller Manager (CCM) handles invalid annotations by producing errors and events, it is difficult 
+   for the Ingress Operator to decipher these CCM events after the load balancer has been provisioned. 
```

Comment:
"it is difficult" doesn't say much, and stands out as a flaw.  Can it be reworded as:

```suggestion
   While the AWS Cloud Controller Manager (CCM) handles invalid annotations by producing errors and events, it doesn't provide enough information 
   for the Ingress Operator to decipher these CCM events after the load balancer has been provisioned. 
```
```suggestion
   While the AWS Cloud Controller Manager (CCM) handles invalid annotations by producing errors and events, it is difficult 
   for the Ingress Operator to decipher these CCM events after the load balancer has been provisioned. 
```